src/components/cells/cell-menu.jsx

```diff
@@ -91,7 +91,8 @@ export class CellMenuUnconnected extends React.Component {
 
 export function mapStateToProps(state, ownProps) {
   const { cellId } = ownProps
-  const userLanguages = Object.keys(state.languages).filter(language => language !== 'js')
+  let userLanguages = Object.assign({}, state.languageDefinitions, state.loadedLanguages)
+  userLanguages = Object.keys(userLanguages).filter(language => language !== 'js')
```

Comment:
perhaps if we're going this way, it makes just as much sense to remove this line and remove the special case NotebookMenuItem for js? won't that be generated automatically by the same mechanism that generates menu items for all the other languages?

Reply:
Yes.  The only advantage here is that `js` will always appear first (which I think has some, if small, merit).